tdrs-backend/tdpservice/parsers/parse.py

```diff
@@ -61,27 +41,69 @@ def parse_datafile(datafile):
         errors['document'] = ['Section does not match.']
         return errors
 
-    line_errors = parse_datafile_lines(datafile, program_type, section)
+    line_errors = parse_datafile_lines(rawfile, program_type, section)
 
     errors = errors | line_errors
 
     return errors
 
 
-def parse_datafile_lines(datafile, program_type, section):
+def store_record(unsaved_records, record, model):
+    """Store record in dictionary for later processing."""
+    if record:
+        if model not in unsaved_records:
+            unsaved_records[model] = [record]
```

Comment:
Nit: to make it more readable, always prefer to do something like:

unsaved_records.update({
    model: record
    })